pandas/tests/io/test_clipboard.py

```diff
@@ -243,6 +243,17 @@ def test_read_clipboard_infer_excel(self, request, mock_clipboard):
 
         tm.assert_frame_equal(res, exp)
 
+    # Tests that nulls on the first column do not trip infering excel format
+    def test_infer_excel_with_nulls(self, request, mock_clipboard):
+        # GH41108
+        text = "col1\tcol2\n1\tred\n\tblue\n2\tgreen"
+
+        mock_clipboard[request.node.name] = text
+        df = read_clipboard()
+
+        # excel data is parsed correctly
+        assert df.iloc[1][1] == "blue"
```

Comment:
please test whole frame, not only one cell.

Reply:
just added an extra check on the whole frame

Reply:
Please use tm.assert_frame_equal,

You can remove the iloc check

Reply:
sorry didn't know about that, just changed it, thanks